rllib/offline/dataset_reader.py

```diff
@@ -251,12 +252,34 @@ def next(self) -> SampleBatchType:
             d = next(self._iter).as_pydict()
             # Columns like obs are compressed when written by DatasetWriter.
             d = from_json_data(d, self._ioctx.worker)
-            d = postprocess_actions(d, self._ioctx)
+            d = postprocess_actions(d, self._ioctx.worker)
             count += d.count
+            d = self._preprocess_if_needed(d)
             ret.append(self._postprocess_if_needed(d))
         ret = concat_samples(ret)
         return ret
 
+    def _preprocess_if_needed(self, batch: SampleBatchType) -> SampleBatchType:
+        # TODO: @kourosh, preprocessor is only supported for single agent case.
```

Comment:
seems like I can replace all of these with connectors after it's enabled by default.

Reply:
sg, this was not supported properly yet. So let's continue with this solution for now.